Sources/NIO/Channel.swift

```diff
@@ -362,3 +362,13 @@ public enum ChannelEvent: Equatable {
     /// Output portion of the `Channel` was closed.
     case outputClosed
 }
+
+/// A `Channel` user event that is sent when the `Channel` has been asked to quiesce.
+///
+/// The action(s) that should be taken after receiving this event are both application and protocol dependent. If the
+/// protocol supports a notion of requests and responses, it might make sense to stop accepting new requests but finish
+/// processing the request currently in flight.
+public struct ChannelShouldQuiesceUserEvent {
```

Comment:
nit: consider using an enum ?

Reply:
@normanmaurer we need to be able to instanciate that one. So I could do an enum but what would the cases be? I could only think of one case "please quiesce the channel" and I thought that's easier expressed by just using a type. But happy to change. We can't use `ChannelEvent` because it's an `enum` and extending it would be a major version break